### PR TITLE
WIP: Dockerify2

### PR DESCRIPTION
**Pull request by revolunet**
Quelques fixes :
 - un probleme d'url dans SearchResults (missing `/`prefix) ?
 - une typo dans l'instantiation piwik
 - correction des routes pour les conventions collectives (en attente de unify-routes)

Et surtout un setup docker+lerna qui fonctionne mais est d'une lenteur extrême.

 - le `.env` et les `Dockerfile` sont à la racine

### CODE REVIEW

File: Dockerfile.python [hunk 1]
```diff
@@ -16,12 +19,8 @@ RUN apk update \
...
-
-WORKDIR $APP_DIR
+COPY ./packages/code-du-travail-data/Pipfile .
 
 RUN pipenv install --deploy
 
-COPY . $APP_DIR
-
-ENTRYPOINT ["/app/docker/entrypoint.sh"]
+COPY ./packages/code-du-travail-data .
```
Comment (lionelB) [inline Dockerfile.python:26]: le fichier docker/python-entrypoint.sh n'est plus appelé ?

File: packages/code-du-travail-api/src/server/index.js [hunk 2]
```diff
@@ -1,38 +1,39 @@
 // Set up Elastic APM agent: the agent must be started before any other modules.
 // https://www.elastic.co/guide/en/apm/agent/nodejs/current/koa.html
-require('elastic-apm-node').start({
```
Comment (douglasduteil) [inline packages/code-du-travail-api/src/server/index.js:3]: Those changes shouldn't be part of this PR

**Review by douglasduteil — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -65,7 +64,18 @@ $ yarn workspace <package-name> test
 
 # For example for "code-du-travail-frontend"
 $ yarn workspace code-du-travail-frontend test
+```
+
+### Prod
+
+Edit `.env`
```
Comment (lionelB) [inline README.md:71]: I would say 

```sh 
$ cp .env.sample .env
```
And edit the .env file.

Comment (douglasduteil) [reply README.md:71]: Isn't is automatique on "postinstall" ?

File: Dockerfile.frontend [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+# TODO: we should switch to https://github.com/zeit/next.js/tree/master/examples/with-universal-configuration-runtime ?
+ARG API_URL
+ARG INTERNAL_API_URL
+ARG SENTRY_PUBLIC_DSN
+ARG PIWIK_URL
+ARG PIWIK_SITE_ID
+
+ENV NODE_ENV=production
+ENV API_URL=$API_URL
+ENV INTERNAL_API_URL=$INTERNAL_API_URL
```
Comment (lionelB) [inline Dockerfile.frontend:30]: what 's the difference between `API_URL` and `INTERNAL_API_URL` ?

**Comment by douglasduteil** [comment]
Wait @revolunet why did you close your PR ?

**Comment by revolunet** [comment]
will be back soon !

File: Dockerfile.frontend (hunk 4 above)
Comment (revolunet) [reply Dockerfile.frontend:30]: yep INTERNAL will not exist but let me explain where it came from as i wasted waaay too much time on this :)

We're currently using next.js which does SSR. 

When the user request some content client-side, like autosuggest, he uses his browser `fetch()` and the public `process.env.API_URL` to resolve the data from our API. all fine.

But when the user request a *page* of content like `/question/how-are-you-today` then that's the server who make the initial request to the API and inject content into the page before returning it to user. this happens in the pages `getInitialProps()` function, which also use `process.env.API_URL`.

So at some point, the server use the public API_URL to make requests to the API server which in fact stands on the same docker-compose environment. 

https://github.com/zeit/next.js/#fetching-data-and-component-lifecycle

Comment (revolunet) [reply Dockerfile.frontend:30]: Also these variables are used at build time and i think we should make them overridable at runtime

File: Dockerfile.python (hunk 1 above)
Comment (revolunet) [reply Dockerfile.python:26]: si je ne m'abuse il ne servait a rien a part poller sur l'elastic search

**Comment by lionelB** [comment]
I've made a PR with getConfig... hope it will return soon :)
